doc/2/guides/main-concepts/1-api/index.md

```diff
@@ -8,3 +8,461 @@ order: 100
 
 # API
 
+Kuzzle exposes most of its features through a **multi-protocol API**.  
+
+This API uses the **JSON format** to communicate with a **standardized request and response format**.
+
+## Multi Protocol
+
+The Kuzzle API is accessible by default through 3 protocols:
+ - [HTTP](/core/2/api/api-protocols/1-http)
+ - [WebSocket](/core/2/api/api-protocols/2-websocket)
+ - [MQTT](/core/2/api/api-protocols/3-mqtt)
+
+Each protocol has advantages and disadvantages. The choice of a protocol must therefore be adapted to a situation and a use.
+
+::: info
+Kuzzle is able to integrate to its API any protocol operating on [IP](https://en.wikipedia.org/wiki/Internet_Protocol).  
+More info on [Writing Protocol Plugin](/core/2/guides/write-plugins/4-network-protocol).  
+:::
+
+## Request Format
+
+Except for HTTP, Kuzzle expects the exact same request format for all communication protocols.
+
+### HTTP
+
+HTTP requests are **split into the four HTTP usual parts**: URL, verb, headers and body.
+
+Every API action documentation has a dedicated HTTP section, explaining how to use that action with the HTTP protocol.
+
+::: info
+You can add the `pretty` parameter in any HTTP URL to receive a pretty formatted JSON response. (e.g. `GET "http://localhost:7512?pretty"`)
+:::
+
+#### Optional Headers
+
+The following list of HTTP headers can be added to any and all HTTP requests:
+
+- `Accept-Encoding`: compression algorithm(s) usable by Kuzzle to encode the response. Accepted encodings, in order of preference: `gzip`, `deflate`, `identity`.
+- `Authorization` (expected value: `Bearer <token>`): user's authentification token, obtained through the [auth:login](/core/2/api/controllers/auth/login) API action
+- `Content-Encoding`: compression algorithm(s) used to encode the body sent to Kuzzle. Accepted encodings: `deflate`, `gzip`, `identity`
+
+#### Body Encoding
+
+Body contents can be sent in the following formats:
+
+- `application/json`: raw JSON
+- `multipart/form-data`: HTML forms; both field-value pairs and field-files pairs can be sent that way
+
+If a HTML form is sent that way, the **resulting body content will be translated into a JSON object**, with as many keys as the provided form fields.  
+If the form field holds a file, then the corresponding JSON key will refer to an object instead of a mere value, with the following properties:
```

Comment:
```suggestion
If the form field holds a file, then the corresponding JSON key will refer to an object instead of a simple value, with the following properties:
```